boards/shields/huatian_mt9m114/huatian_mt9m114.overlay

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/{
+	chosen {
+		zephyr,camera = &nxp_csi;
+	};
+};
+
+&zephyr_cam_i2c {
+	mt9m114: mt9m114@48 {
+		compatible = "aptina,mt9m114";
+		reg = <0x48>;
+
+		port {
+			mt9m114_ep_out: endpoint {
+				remote-endpoint = <&csi_ep_in>;
+			};
+		};
+	};
+};
+
+&nxp_csi {
```

Comment:
As you said, that's what I understand about a shield, it should be generic. But CSI is a specific IP of NXP and its node contains the property "sdev" (source) which is used to reference to a camera sensor.

So I don't know if we could generalize the name to `fpc_sensor_interface` ? similar to what I answered to @danieldegrasse below.

Otherwise, @danieldegrasse and @mmahadevan108 commented that things specific to NXP should be named "nxp_" (e.g. nxp_cam_i2c, nxp_csi, etc.). And in practice, there are shields that defined like that in Zephyr, such as **rk055hdmipi4ma0** ... I am confused ...

[danieldegrasse](https://github.com/danieldegrasse) [2 weeks ago](https://github.com/zephyrproject-rtos/zephyr/pull/72435#discussion_r1596039125)
Perhaps we should label this as nxp_csi? These properties aren't generic between all CSI devices in Zephyr, are they?

Contributor
Author
@[ngphibang](https://github.com/ngphibang) ngphibang [last week](https://github.com/zephyrproject-rtos/zephyr/pull/72435#discussion_r1597370704) • 
The property "sdev" here is not generic. But the purpose of this labelling "zephyr_xxx" is for other boards / vendors which have similar camera pipeline, e.g. "mt9m114->csi" (a sensor connected to a sensor interface is a quite common pipeline) can reuse the overlay. And to be able to use the overlay, they should use the same property "sdev" in their devicetree ?

That is my understanding about the main purpose / benefit of making a "shield", no ? If all things in a shield are just specific to NXP, so how can other boards / vendors use it ?

BTW, the name csi seems specific to NXP (and also not a well chosen name at all as it is confused with "MIPI CSI"). We could re-label it to "zephyr_sensor_interface" so that all boards / vendors who have the same camera pipeline "mt9m114 -> sensor_interface" can reuse this overlay.

Reply:
> But CSI is a specific IP of NXP and its node contains the property "source" which is used to reference to a camera sensor.

Then this part should probably be in a `boards/shields/huatian_mt9m114/boards/xyz.overlay` file specific to the particular NXP board(s) for which you want to set the source property of the csi node.

Reply:
Yeah, it's a solution. But if we cut and put this into a board specific overlay, there's nearly nothing left in the shield :).

My point is that a 24-pin connector camera interface (such as the NXP's CSI) definitly has to connect to a camera sensor (mt9m114). So the reference to a the mt9m114 is always needed. Here, it's just the name of the property `source` is NXP's definition (in CSI's dt binding). Can we still keep this node in the shield and if someone want to reuse the overlay, they just needs to rename their camera interface 's property to source ?

```
&fpc_cam_interface {
	status = "okay";
	source = <&mt9m114>;

	port {
		fci_ep_in: endpoint {
			remote-endpoint = <&mt9m114_ep_out>;
		};
	};
};
```